pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -226,6 +226,23 @@ func (r *ReconcileLogCollector) Reconcile(request reconcile.Request) (reconcile.
 		}
 	}
 
+	var splunkCredential *render.SplunkCredential
+	if instance.Spec.AdditionalStores != nil {
+		if instance.Spec.AdditionalStores.Splunk != nil {
+			splunkCredential, err = getSplunkCredential(r.client)
+			if err != nil {
+				log.Error(err, "Error with Splunk credential secret")
+				r.status.SetDegraded("Error with Splunk credential secret", err.Error())
+				return reconcile.Result{}, err
+			}
+			if splunkCredential == nil {
```

Comment:
Is this required for all Splunk setups?
I don't know splunk config so this may be required, I just wanted to confirm.

Reply:
@tmjd not sure.. can you elaborate the comment ?

Reply:
Is the splunkCredential required for all splunk endpoints? With this check, progress is blocked without the credential, is that expected?

Reply:
looking at getSplunkCredential is looks like the token is required but not the CA.

Reply:
Yes, once the user input the `SplunkSpec` in the `AdditionalStores` we require the token to be present, we don't expect the CA certificate. In `getSplunkCredential` we are checking for all the error cases are returning `nil` whenever there is an error case. 

Just so I get this correctly do we want to change the above to this ? This check is already done in the `getSplunkCredential`

```
if len(splunkCredential.Token) == 0 {
}
```

Reply:
I don't think any change is needed. Token is required but CA is not, and the getSplunkCredential takes care of checking that.
All good. :+1: